morf-oracle/src/main/java/org/alfasoftware/morf/jdbc/oracle/OracleDialect.java

```diff
@@ -1123,6 +1126,133 @@ public Collection<String> addTableFromStatements(Table table, SelectStatement se
+
+    // Start by checking if this is a distinct call, then add the field list
+    if (stmt.isDistinct()) {
+      result.append("DISTINCT ");
+    }
+    if (stmt.getFields().isEmpty()) {
+      result.append("*");
+    } else {
+      result.append(getFieldsForSelect(table, stmt));
+    }
+
+    result.append(appendPostSelectSQL(stmt));
+
+    // Any hint directives which should be inserted right at the end of the statement
+    result.append(selectStatementPostStatementDirectives(stmt));
+
+    return result.toString();
+  }
+
+  /**
+   * Returns fields to be used in the select statement.
+   * @param table table definition for the table being created.
+   * @param stmt the select statement.
+   * @return returns all the fields to be included in the select statement in SQL format.
+   */
+  private String getFieldsForSelect(Table table, SelectStatement stmt) {
+    StringBuilder stringBuilder = new StringBuilder();
+    boolean firstField = true;
+    for (AliasedField currentField : stmt.getFields()) {
+      if (!firstField) {
+        stringBuilder.append(", ");
+      }
+
+      if (currentField instanceof NullFieldLiteral) {
+        stringBuilder.append(getSqlForNullField(currentField, table, stmt.getFields()));
+      } else if (currentField instanceof FieldReference) {
+        stringBuilder.append(getSqlField(currentField, table, stmt.getFields()));
+      } else {
+        stringBuilder.append(getSqlFrom(currentField));
+      }
+      // Put an alias in, if requested
+      appendAlias(stringBuilder, currentField);
+
+      firstField = false;
+    }
+    return stringBuilder.toString();
+  }
+
+  /**
+   * Creates the Cast SQL for a given null field.
+   * @param currentField - will be a NullFieldLiteral.
+   * @param table - Table definition for the table being created.
+   * @param fields - All the fields being added to the select.
+   * @return String value of the CAST.
+   */
+  private String getSqlForNullField(AliasedField currentField, Table table, List<AliasedField> fields) {
+    Column column = table.columns().get(fields.indexOf(currentField));
+    return  "CAST(null AS "
+                + sqlRepresentationOfColumnType(column, false, false, true)
+                + ") AS "
+                + column.getName();
+  }
+
+  /**
+   * Creates the Cast SQL for a given field.
+   * @param currentField - will be a FieldReference.
+   * @param table - Table definition for the table being created.
+   * @param fields - All the fields being added to the select.
+   * @return String value of the CAST.
+   */
+  private String getSqlField(AliasedField currentField, Table table, List<AliasedField> fields) {
+    Column column = table.columns().get(fields.indexOf(currentField));
+
+    return  "CAST("
+                + column.getName()
+                + " AS "
+                + sqlRepresentationOfColumnType(column, false, false, true)
+                + ") AS "
+                + column.getName();
+  }

```

Comment:
the fist `column.getName()` should be the SQL of the `currentField` instead.

Reply:
@jsimlo I have fixed this